### PR TITLE
Re-opened: Setup Typescript Bull Redis Queue

### PR DESCRIPTION
**Pull request by jkeat**
Links
-----
- Issue: https://upchieve.monday.com/boards/438837796/pulses/518195818
- Redis: https://redis.io/topics/quickstart
- Bull: https://github.com/OptimalBits/bull

Description
-----------
- adds a typescript bull worker queue
- redis db expected to be local
- adds a new `.eslintrc.js` config to handle typescript files, moves the current rules to `eslint-javascript.js`
- updates `dbconnect` to return a promise as well as take a callback, also uses the default mongoose connection

Developer self-review checklist
-------------------------------
- [ ] Potentially confusing code has been explained with comments
- [ ] No warnings or errors have been introduced; all known error cases have been handled
- [ ] Any appropriate documentation (within the code, README.md, docs, etc) has been updated
- [ ] All edge cases have been addressed

### CODE REVIEW

File: worker/jobs/updateElapsedAvailability.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { log } from '../logger';
+
+export default async (): Promise<void> => {
+  try {
+    await dbconnect();
+    // Fetch volunteers
+    const volunteers = (await UserModel.find({
+      isVolunteer: true
+    }).exec()) as User[];
+    await Promise.all(
```
Comment (ericyliu) [inline worker/jobs/updateElapsedAvailability.ts:15]: promise.all seems to work, but interestingly enough, the difference between doing these in sequence and parallel time-wise is only 2 seconds..., 1:23 vs 1:25

Comment (ericyliu) [reply worker/jobs/updateElapsedAvailability.ts:15]: perhaps we only have 1 connection to mongo open through mongoose and its doing all the saves in sequence anyways

**Comment by ericyliu** [comment]
@jkeat after merging master in this branch, im now getting a `Cannot read property 'match' of undefined` when trying to run `npm install`. Its also the reason circle is failing

**Comment by jkeat** [comment]
LGTM 👍

**Pull request merged**